VideoLocker/src/main/java/org/edx/mobile/view/CourseDiscussionPostsThreadFragment.java

```diff
@@ -293,11 +303,26 @@ public void onException(Exception ex) {
         getThreadListTask.execute();
     }
 
-    private void checkNoResultView() {
+    private void checkNoResultView(EmptyQueryResultsFor query) {
         Activity activity = getActivity();
         if (activity instanceof TaskProcessCallback) {
             if (discussionPostsAdapter.getCount() == 0) {
-                String resultsText = getActivity().getResources().getString(R.string.forum_empty);
+                String resultsText = "";
+                switch (query) {
+                    case FOLLOWING:
+                        resultsText = getActivity().getResources().getString(R.string.forum_no_results_for_following);
+                        break;
+                    case CATEGORY:
+                        resultsText = getActivity().getResources().getString(R.string.forum_no_results_in_category);
+                        break;
+                    case COURSE:
+                        resultsText = getActivity().getResources().getString(R.string.forum_no_results_for_all_posts);
+                        break;
+                }
+                if (postsFilter != DiscussionPostsFilter.ALL) {
+                    resultsText += " " + getActivity().getResources().getString(R.string.forum_no_results_with_filter);
```

Comment:
This is not RTL language friendly.

Perhaps use a placeholder in `forum_no_results_with_filter` instead of string concatenation? e.g. "{reason}\nTry removing your filter"